File: minesweeper/java-minesweeper/src/main/java/it/chicio/minesweeper/field/parser/FieldRowContentTerminationChainParser.java
package it.chicio.minesweeper.field.parser;

import it.chicio.minesweeper.field.Field;

public class FieldRowContentTerminationChainParser extends FieldRowContentChainParser {
    boolean canParse(String row) {
        return isTermination(row);
    }

    private boolean isTermination(String row) {
        return row.equals("0 0");
    }

    public FieldsParsingStatus parse(FieldsParsingStatus fieldsParsingStatus) throws RuntimeException {
        FieldsParsingStatus newFieldsParsingStatus = new FieldsParsingStatus(fieldsParsingStatus);
        if (isValid(newFieldsParsingStatus.currentField)) {
            newFieldsParsingStatus.fieldsParsed.add(newFieldsParsingStatus.currentField);
        }
        return newFieldsParsingStatus;
    }

    private boolean isValid(Field field) {
        return field != null;
    }
}

File: minesweeper/kotlin-mockk-minesweeper/src/main/kotlin/it/chicio/minesweeper/field/parser/FieldRowContentTerminationChainParser.kt
package it.chicio.minesweeper.field.parser

class FieldRowContentTerminationChainParser : FieldRowContentChainParser() {
    override fun canParse(row: String?): Boolean = isTermination(row)

    private fun isTermination(row: String?): Boolean = row == "0 0"

    override fun parse(fieldsParsingStatus: FieldsParsingStatus): FieldsParsingStatus = fieldsParsingStatus
            .currentField
            ?.let {
                val newFieldParsingStatus = fieldsParsingStatus.copy()
                newFieldParsingStatus.fieldsParsed.add(it)
                newFieldParsingStatus
            } ?: fieldsParsingStatus.copy()
}